File: app/src/main/java/pomis/app/telenote/networking/Endpoints.java
package pomis.app.telenote.networking;

import org.json.JSONObject;

import io.reactivex.Observable;
import pomis.app.telenote.models.Page;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by romanismagilov on 06.06.17.
 */

public interface Endpoints {

    @GET("/getPageList")
    Observable<JSONObject> getPageList(
            @Query("access_token") String accessToken,
            @Query("limit") int limit
    );
}

File: app/src/main/java/pomis/app/telenote/api/Endpoints.kt
package pomis.app.telenote.api

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

interface Endpoints {

    @GET("getPageList?access_token={accessToken}&limit={limit}")
    fun getPageList(
            @Path("accessToken") accessToken: String,
            @Path("limit") limit: Int
    ): Observable<String>
}